### PR TITLE
fix compatibility issues with cucumber 7.3.0

### PR DESCRIPTION
**Pull request by davidjgoss**
Fixes a runtime issue with the newest version of cucumber-js after some changes to messages.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,25 +35,26 @@
...
-    "@cucumber/messages": "^13.2.1",
     "ansi-styles": "^5.0.0",
     "cli-table3": "^0.6.0",
     "figures": "^3.2.0",
     "ts-dedent": "^2.0.0"
   },
   "peerDependencies": {
-    "@cucumber/cucumber": ">=7.0.0"
+    "@cucumber/cucumber": ">=7.0.0",
+    "@cucumber/messages": "*"
```
Comment (davidjgoss) [inline package.json:45]: `messages` should be fine as a peer and dev dependency - we know `cucumber` will always bring it along, and it's only needed here for TypeScript reasons - once we compile down to JS there is no import of `messages`.

File: src/index.ts [hunk 2]
```diff
@@ -182,7 +183,7 @@ export default class PrettyFormatter extends SummaryFormatter {
       (item) => item.id === testStepStarted.testStepId
     )
 
-    if (testStep && testStep.pickleStepId !== '') {
```
Comment (davidjgoss) [inline src/index.ts:185]: This is the only change of substance beyond TypeScript compilation stuff, and what is currently broken with 7.3.0. Example:

```
TypeError: Cannot read property 'astNodeIds' of undefined
    at PrettyFormatter.onTestStepStarted (/Users/davidgoss/Projects/cucumber-puppeteer-typescript-starter/node_modules/@cucumber/pretty-formatter/src/index.ts:187:36)
    at PrettyFormatter.parseEnvelope (/Users/davidgoss/Projects/cucumber-puppeteer-typescript-starter/node_modules/@cucumber/pretty-formatter/src/index.ts:137:12)
    at EventEmitter.emit (events.js:326:22)
```

With the change in `messages` from using protobuf to JSON schema, some fields that used to be defaulted to an empty string (per protobuf convention) are now undefined if not set. Making this a simple truthiness check solves the issue and is still backwards-compatible with the older version.

File: src/index.ts [hunk 3]
```diff
@@ -241,14 +241,11 @@ export default class PrettyFormatter extends SummaryFormatter {
...
-  private onTestCaseFinished(_testCaseFinished: messages.ITestCaseFinished) {
+  private onTestCaseFinished(_testCaseFinished: messages.TestCaseFinished) {
     this.newline()
   }
 
-  private renderTags(
-    indent: number,
-    tags: messages.GherkinDocument.Feature.ITag[]
-  ) {
+  private renderTags(indent: number, tags: readonly { name: string }[]) {
```
Comment (davidjgoss) [inline src/index.ts:248]: Just had to change this `tags` parameter to the bare minimum interface shared by `Tag` and `PickleTag`

File: src/index.ts [hunk 4]
```diff
@@ -141,7 +142,7 @@ export default class PrettyFormatter extends SummaryFormatter {
       this.onTestCaseFinished(envelope.testCaseFinished)
   }
 
-  private onTestCaseStarted(testCaseStarted: messages.ITestCaseStarted) {
+  private onTestCaseStarted(testCaseStarted: messages.TestCaseStarted) {
```
Comment (jbpros) [inline src/index.ts:145]: Ha, this is much better :)

**Review by jbpros — APPROVED** [review]
Thank you, this is great. I like how we got rid of those interfaces, too.

**Comment by davidjgoss** [comment]
Thanks @jbpros!

**Pull request merged**